exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt

```diff
@@ -184,7 +184,25 @@ object SchemaUtils {
                         is MysqlDialect -> value.setScale((exp.columnType as DecimalColumnType).scale).toString()
                         else -> processForDefaultValue(exp)
                     }
-                    else -> processForDefaultValue(exp)
+                    else -> {
+                        if (column.columnType is IJsonColumnType) {
+                            val processed = processForDefaultValue(exp)
+                            when (dialect) {
+                                is PostgreSQLDialect -> {
+                                    if (column.columnType.usesBinaryFormat) {
+                                        processed.replace(Regex("(\"|})(:|,)(\\[|\\{|\")"), "$1$2 $3")
+                                    } else {
+                                        processed
+                                    }
+                                }
```

Comment:
This is necessary because PostgreSQL metadata returns JSONB column values with added whitespace, but it still does not guarantee an accurate comparison if a complex/deeply nested json string is used as a default (hopefully an edge case).

This is because JSONB also does not retain order of key-value pairs and will remove duplicates too, so a simple string comparison won't be enough.

A thorough comparison check may need to be investigated if users log issues with jsonb defaults and `createMissingTablesAndColumns()` in PostgreSQL (also dependent on the future of ALTER TABLE functionality).